packages/transitive-overlay/src/route-layers.ts

```diff
@@ -32,16 +39,58 @@ export function patternToRouteFeature(
     .replace(/\s/g, "!")
     .replace(/[+-0124679FHJLPTVXYZ]/g, "E");
 
+  const properties = {
+    color: `#${route.route_color || "000080"}`,
+    name: routeName,
+    nameUpper: routeNameUpper,
+    routeType: route.route_type,
+    textColor: `#${route.route_text_color || "eee"}`,
+    type: "route"
+  };
+
+  const isFlex = pattern.stops[pattern.stops.length - 1].stop_id.includes(
+    "flexed_to"
+  );
+  const straight = polyline.toGeoJSON(polyline.encode(concatenatedLines));
+
+  if (isFlex) {
+    // Create clone of plain route that only includes first and last point
+    const extraStraight = { ...straight };
+    extraStraight.coordinates = [
+      extraStraight.coordinates[0],
+      extraStraight.coordinates[extraStraight.coordinates.length - 1]
```

Comment:
Maybe extract `const`s for these two coordinates, they are used often by themselves.

Reply:
Much cleaner, thanks!